src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -594,7 +594,7 @@ abstract class RefChecks extends Transform {
         def checkNoAbstractMembers(): Unit = {
           // Avoid spurious duplicates: first gather any missing members.
           def memberList = clazz.info.nonPrivateMembersAdmitting(VBRIDGE)
-          val (missing, rest) = memberList partition (m => m.isDeferred && !ignoreDeferred(m))
+          val missing = memberList.reverseIterator.filter(m => m.isDeferred && !ignoreDeferred(m)).toList.reverse
```

Comment:
Maybe we can avoid even more work with something like...

```scala
          val (missing, rest) = {
            val missing = ArrayBuffer[global.Symbol]()
            val rest = ArrayBuffer[global.Symbol]()

            memberList.reverseIterator.foreach { s =>
              if (s.isDeferred && !ignoreDeferred(s)) {
                missing.addOne(s)
              } else if (s.isAbstractOverride && s.isIncompleteIn(clazz)) {
                rest.addOne(s)
              }
            }

            (missing, rest)
          }
```

...

```scala
rest.reverseIterator.foreach { member => 
  ...
```
...
```scala
          for (member <- missing.reverseIterator ; msg = diagnose(member) ; if msg != null) {
...
```

Reply:
So, I did not thought at first about optimising this part because these are lists of errors, which are always empty for correct compilations. 

What I have done at the end is to use mutable `List` variables, and use the `::=` operator for the LIFO behaviour. This has the benefit of having no allocations (`Nil`) in the common, no-errors case.

Reply:
It may be possible to also add the `grouped` and `missingMethods` logic, for consolidating pairs of setter-getter into a single element, but I am not sure it would be worth as much.